include/problem/MFEMProblem.h

```diff
@@ -154,6 +154,33 @@ class MFEMProblem : public ExternalProblem
   void setMOOSENodalVarData(MooseVariableFieldBase & moose_variable);
   void setMOOSEElementalVarData(MooseVariableFieldBase & moose_variable);
 
+  /**
+   * Template method for adding kernels. We can only add kernels using equation system problem builders.
+   */
+  template <class T>
+  void addKernel(std::string var_name, std::shared_ptr<hephaestus::Kernel<T>> kernel)
+  {
+    using namespace hephaestus;
+
+    TimeDomainEquationSystemProblemBuilder * td_eqn_system_problem_builder{nullptr};
+    SteadyStateEquationSystemProblemBuilder * ss_eqn_system_problem_builder{nullptr};
+
+    ProblemBuilder * problem_builder = mfem_problem_builder.get();
+
+    if ((td_eqn_system_problem_builder = dynamic_cast<TimeDomainEquationSystemProblemBuilder *>(problem_builder)))
```

Comment:
Could just cast to the `EquationSystemProblemBuilderInterface` that gives access to AddKernel?